students/mayc4t/lesson07/html_render_test.py

```diff
@@ -0,0 +1,422 @@
+#!/usr/bin/env python3
+
+import os
+import unittest
+import html_render
+
+CLEANUP_TEST_FILES=True
+
+def validate_output(file_out, want_lines):
+    """validate_output parses a rendered file and validates its content."""
+
+    # Extract got from file_out.
+    got_lines = []
+    try:
+        f = open(file_out)
+        got_lines = f.readlines()
+        f.close()
+    except:
+        print('validate_output failed to validate %s: %s' %
+              (file_out, sys.exc_info()[0]))
+        return False
+
+    # Check line counts.
+    got_line_cnt = len(got_lines)
+    want_line_cnt = len(want_lines)
+    if (got_line_cnt != want_line_cnt):
+        print('FAIL: %s has %s lines, expected %s lines.' %
+              (file_out, got_line_cnt, want_line_cnt))
+        return False
+
+    # Check each line.
+    good = True
+    line_num = 0
+    for got_line, want_line in zip(got_lines, want_lines):
+        got_line = got_line.rstrip('\n')
+        if (got_line != want_line) :
+            print('FAIL at line %s: got \'%s\', expected \'%s\'' %
+                  (line_num, got_line, want_line))
+            good = False
+        line_num += 1
+
+    return good
+
+def rm_test_file(file_out):
+    try:
+        if (CLEANUP_TEST_FILES):
+            os.remove(file_out)
+    except:
+        pass
+
+class TestTextWrapper(unittest.TestCase):
+    def setUp(self):
+        """setUp creates a file to back the TextWrapper class."""
+        self.file_out = 'text_wrapper_test.txt'
```

Comment:
you can use `StringIO` module which gives you a file like object without creating actual files.